### PR TITLE
Update to use the new hc-releases

### PR DESCRIPTION
**Pull request by jeanneryan**
Add necessary components to onboard to Releases API

### CODE REVIEW

File: .github/workflows/hashicorp.yml [hunk 1]
```diff
@@ -57,6 +46,19 @@ on:
...
+        required: false
+      hc-releases-key-prod:
+        description: 'Key for hc-releases prod'
+        required: false
+      hc-releases-host-prod:
+        description: 'Host for hc-releases prod'
+        required: false
+      hc-releases-host-staging:
+        description: 'Host for hc-releases staging'
+        required: false
```
Comment (bflad) [inline .github/workflows/hashicorp.yml:60]: Should we mark these as required and update the descriptions with the typical organization/repository secret names? We'll also need to update the README and engineering-docs when things are ready. 😄 

```suggestion
      hc-releases-key-staging:
        description: 'Key for hc-releases staging (HC_RELEASES_KEY_STAGING)'
        required: true
      hc-releases-key-prod:
        description: 'Key for hc-releases prod (HC_RELEASES_KEY_PROD)'
        required: true
      hc-releases-host-prod:
        description: 'Host for hc-releases prod (HC_RELEASES_HOST_PROD)'
        required: true
      hc-releases-host-staging:
        description: 'Host for hc-releases staging (HC_RELEASES_HOST_STAGING)'
        required: true
```

File: .github/workflows/hashicorp.yml [hunk 2]
```diff
@@ -28,19 +28,12 @@ on:
         default: 'signing_subkey_CD27AB87'
         required: false
         type: string
-    secrets:
-      hc-releases-aws-access-key-id:
```
[the remark this answers is not in the document]
Comment (jeanneryan) [reply .github/workflows/hashicorp.yml:32]: I will do that now, thank you for the quick turn-around!

**Review by detro — APPROVED** [review]
Nothing pops at me, but I did leave some questions, especially around the whole "staging -> metadata -> production" promotion, that I don't fully follow how can be leverage by us.

If there was a way to release in staging our providers, test drive them (with customers?), and then promote to production, I wouldn't be against that :)

File: .github/workflows/hashicorp.yml [hunk 3]
```diff
@@ -70,29 +76,23 @@ jobs:
...
-      - uses: aws-actions/configure-aws-credentials@v1
-        with:
-          aws-access-key-id: ${{ secrets.hc-releases-aws-access-key-id }}
-          aws-secret-access-key: ${{ secrets.hc-releases-aws-secret-access-key }}
-          aws-region: us-west-2
-          role-to-assume: ${{ secrets.hc-releases-aws-role-arn }}
-          role-duration-seconds: ${{ inputs.hc-releases-aws-role-duration-seconds }}
-          # TODO: allow session tagging once IAM permission changes addressed
-          # Reference: https://github.com/hashicorp/hc-releases/issues/124
-          role-skip-session-tagging: true
```
[the remark this answers is not in the document]
Comment (jeanneryan) [reply .github/workflows/hashicorp.yml:89]: Yes, that is correct.

File: .github/workflows/hashicorp.yml [hunk 4]
```diff
@@ -57,6 +50,19 @@ on:
...
+        required: true
+      hc-releases-key-prod:
+        description: 'Key for hc-releases prod (HC_RELEASES_KEY_PROD)'
+        required: true
+      hc-releases-host-prod:
+        description: 'Host for hc-releases prod (HC_RELEASES_HOST_PROD)'
+        required: true
+      hc-releases-host-staging:
+        description: 'Host for hc-releases staging (HC_RELEASES_HOST_STAGING)'
+        required: true
```
[the remark this answers is not in the document]
Comment (jeanneryan) [reply .github/workflows/hashicorp.yml:64]: hi @detro ,  there are two services but only the production service is publicly accessible.

File: .github/workflows/hashicorp.yml [hunk 5]
```diff
@@ -111,14 +111,29 @@ jobs:
...
+      -
+        name: Promote
+        uses: hashicorp/actions-hc-releases-promote@v1
+        with:
+          product-name: ${{ github.event.repository.name }}
+          version: ${{ inputs.product-version }}
+          hc-releases-host: ${{ secrets.hc-releases-host-prod }}
+          hc-releases-key: ${{ secrets.hc-releases-key-prod }}
+          hc-releases-source_env_key: ${{ secrets.hc-releases-key-staging }}
+          hc-releases-terraform-registry-sync-token: ${{ secrets.hc-releases-terraform-registry-sync-token }}
```
[the remark this answers is not in the document]
Comment (jeanneryan) [reply .github/workflows/hashicorp.yml:139]: `hc-releases-source_env_key` is used for authentication/authorization between the production and staging services. It does not affect which release is promoted. 

You could set up a workflow that publishes to staging and don't promote beyond that.

**Comment by detro** [comment]
I have gathered answers to my questions here and in slack, so I resolved what was left.

I'm eager to see this merged and start updating our utility providers @jeanneryan - will this be mergeable within a day or two?

**Pull request merged**